CONTRIBUTING.md

```diff
@@ -1,3 +1,336 @@
+contributing to Helm. All issue types follow the same general lifecycle. Differences are noted
+below.
+
+1. Issue creation
+2. Triage
+    - The maintainer in charge of triaging will apply the proper labels for the issue. This includes
+      labels for priority, type, and metadata (such as `good first issue`). The only issue priority
+      we will be tracking is whether or not the issue is "critical." If additional levels are needed
+      in the future, we will add them.
+    - (If needed) Clean up the title to succinctly and clearly state the issue. Also ensure that
+      proposals are prefaced with "Proposal: [the rest of the title]".
+    - Add the issue to the correct milestone. If any questions come up, don't worry about adding the
+      issue to a milestone until the questions are answered.
+    - We attempt to do this process at least once per work day.
+3. Discussion
+    - Issues that are labeled `feature` or `proposal` must write a Helm Improvement Proposal (HIP).
+      See [Proposing an Idea](#proposing-an-idea). Smaller quality-of-life enhancements are exempt.
+    - Issues that are labeled as `feature` or `bug` should be connected to the PR that resolves it.
+    - Whoever is working on a `feature` or `bug` issue (whether a maintainer or someone from the
+      community), should either assign the issue to themselves or make a comment in the issue saying
+      that they are taking it.
+    - `proposal` and `support/question` issues should stay open until resolved or if they have not
+      been active for more than 30 days. This will help keep the issue queue to a manageable size
+      and reduce noise. Should the issue need to stay open, the `keep open` label can be added.
+4. Issue closure
+
+## Proposing an Idea
+
+Before proposing a new idea to the Helm project, please make sure to write up a [Helm Improvement
+Proposal](https://github.com/helm/community/tree/master/hips). A Helm Improvement Proposal is a
+design document that describes a new feature for the Helm project. The proposal should provide a
+concise technical specification and rationale for the feature.
+
+It is also worth considering vetting your idea with the community via the
+[cncf-helm](mailto:cncf-helm@lists.cncf.io) mailing list. Vetting an idea publicly before going as
+far as writing a proposal is meant to save the potential author time. Many ideas have been proposed;
+it's quite likely there are others in the community who may be working on a similar proposal, or a
+similar proposal may have already been written.
+
+HIPs are submitted to the [helm/community repository](https://github.com/helm/community). [HIP
+1](https://github.com/helm/community/blob/master/hips/hip-0001.md) describes the process to write a
+HIP as well as the review process.
+
+After your proposal has been approved, follow the [developer's
+guide](https://helm.sh/docs/community/developers/) to get started.
```

Comment:
You can drop this section. This plugin isn't going to go outside of scope of it's existing place so something a HIP would deal with won't happen here.